x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/Precision.java

```diff
@@ -140,8 +140,7 @@ public String getName() {
 
     @Override
     public void process(Aggregations aggs) {
-        if (topActualClassNames.get() == null && aggs.get(ACTUAL_CLASSES_NAMES_AGG_NAME) instanceof Terms) {
-            Terms topActualClassesAgg = aggs.get(ACTUAL_CLASSES_NAMES_AGG_NAME);
+        if (topActualClassNames.get() == null && aggs.get(ACTUAL_CLASSES_NAMES_AGG_NAME)instanceof Terms topActualClassesAgg) {
```

Comment:
does the formatter result to no space before `instanceof`s? It seems to be the case when the instanceofs are in if clauses in this PR.

Reply:
The formatter doesn't work with pattern matching instanceof when there's a bracket immediately before the instanceof.  This has been discussed in ES delivery Slack.

Apparently the bug is already fixed in the next version, so we'll get the fix in due course as the build infrastructure gets upgraded.

This of course will cause a problem if this code is committed as-is: it has to have weird formatting to pass CI today but will make CI fail when the build infrastructure is upgraded.

The pragmatic thing to do would be to remove the changes that currently have weird formatting from this PR, then add them back in the future once the formatter is fixed.  Or some of them can be refactored to avoid the bracket next to the instanceof and also avoid a function call.

For example, it's possible to change:

```
    if (object.method() instanceof Something something) {
        // blah
    } else if (object.method() instanceof SomethingElse somethingElse) {
        // blah
    }
```

to:

```
    Thing thing = object.method();
    if (thing instanceof Something something) {
        // blah
    } else if (thing instanceof SomethingElse somethingElse) {
        // blah
    }
```

Maybe there are places where that will make the code much more verbose, e.g. where there are several temporary variables that would need creating.

Reply:
I will look for the weird places like this and see about fixing them pragmatically :)